tensilelite/Tensile/KernelWriterAssembly.py

```diff
@@ -8053,32 +8100,59 @@ def globalWriteElements(self, kernel, tPA, tPB, vectorWidths, elements,
+            module.add(SCSelectB32(dst=sgpr(tmpSgpr), src0=sgpr("SizeI"), src1=sgpr("SizeJ")))
+            module.add(set_bs_label)
+          else:
+            module.add(SCSelectB32(dst=sgpr(tmpSgpr), src0=sgpr("SizeI"), src1=sgpr(tmpSgpr)))
           module.add(allocPostLoopSrdSuppress("Bias", labelStr, sgprLength=sgpr(tmpSgpr)))
-        multiBiasTypeLabel = []
-        for i in kernel["ProblemType"]["BiasDataTypeList"]:
-          name = self.labels.getNameInc("Load_Bias%s"%i.toNameAbbrev())
-          multiBiasTypeLabel.append(Label(name, ""))
+
+        name = self.labels.getNameInc("Load_BiasDim_%u"%(0))
+        biasDim0Label = Label(self.labels.getNameInc("Load_BiasDim_0"), "")
+        biasDim1Label = Label(self.labels.getNameInc("Load_BiasDim_1"), "")
         loadBiasEndLabel = Label(self.labels.getNameInc("Load_Bias_End"), "")
-        multiBiasTypeLabel.append(loadBiasEndLabel)
-        offsetVgpr  = self.vgprPool.checkOut(1, 1)
-        with self.allocTmpSgpr(1, 1) as tmpSgprRes:
-          if len(kernel["ProblemType"]["BiasDataTypeList"]) == 1:
-            module.add(self.readBiasToLDS(kernel["ProblemType"]["BiasDataTypeList"][0], kernel, 1, offsetVgpr, tmpSgprRes.idx, tmpVgprRes))
-          else:
-            for i, label in enumerate(multiBiasTypeLabel[1:]):
-              typeValue = kernel["ProblemType"]["BiasDataTypeList"][i].value
-              module.add(multiBiasTypeLabel[i])
-              module.add(SCmpKLGU32(sgpr("BiasType"), typeValue, "BiasType != %u"%typeValue))
-              module.add(SCBranchSCC1(label.getLabelName(), "Branch if true"))
-              module.add(self.readBiasToLDS(kernel["ProblemType"]["BiasDataTypeList"][i], kernel, 1, offsetVgpr, tmpSgprRes.idx, tmpVgprRes))
-              module.add(SBranch(labelName=loadBiasEndLabel.getLabelName(), comment="Branch to load bias end"))
-            module.add(loadBiasEndLabel)
-        self.vgprPool.checkIn(offsetVgpr)
+        biasDims = [0]
+        if self.states.BiasDim:
+          module.add(biasDim0Label)
+          module.add(SCmpKLGU32(sgpr("BiasDim"), 0, "BiasDim != 0"))
+          module.add(SCBranchSCC1(biasDim1Label.getLabelName(), "Branch if true"))
+          biasDims.append(1)
+
+        for d in biasDims:
+          if d == 1:
+            module.add(biasDim1Label)
+          multiBiasTypeLabel = []
+          for i in kernel["ProblemType"]["BiasDataTypeList"]:
+            name = self.labels.getNameInc("Load_Bias%s_%u"%(i.toNameAbbrev(), d))
+            multiBiasTypeLabel.append(Label(name, ""))
+          multiBiasTypeLabel.append(loadBiasEndLabel)
+          offsetVgpr  = self.vgprPool.checkOut(1, 1)
+          with self.allocTmpSgpr(1, 1) as tmpSgprRes:
+            if len(kernel["ProblemType"]["BiasDataTypeList"]) == 1:
+              module.add(self.readBiasToLDS(kernel["ProblemType"]["BiasDataTypeList"][0], kernel, 1, offsetVgpr, tmpSgprRes.idx, tmpVgprRes, d))
+              if len(biasDims) == 2:
+                if d == 0:
```

Comment:
place the end label when the last bias dim finished otherwise add branch instructions for branching to the end label.